lib/jekyll/drops/drop.rb

```diff
@@ -138,6 +138,15 @@ def inspect
         JSON.pretty_generate to_h
       end
 
+      # When a JSON representation of the drop is requested (e.g. through
+      # template filters) then only show the YAML front matter data.
+      # Other Jekyll objects might not be dumpable due to recursiveness.
+      #
+      # Returns a JSON dump of the YAML front matter data.
+      def to_json
+        fallback_data.to_json
```

Comment:
This feels incomplete to me... We definitely need more than just the fallback data to be jsonified. Ideally we just call `.inspect` on the values in `next` and `previous`.